internal/proxy/proxy.go

```diff
@@ -323,14 +490,35 @@ func (m MultiErr) Error() string {
 }
 
 func (c *Client) Close() error {
+	mnts := c.mnts
+
+	c.fuseServerMu.Lock()
+	hasFuseServer := c.fuseServer != nil
+	c.fuseServerMu.Unlock()
+
 	var mErr MultiErr
+	if hasFuseServer {
+		if err := c.fuseServer.Unmount(); err != nil {
```

Comment:
You have created the infamous double-check-lock pattern here. 

This line probably needs to be protected by c.fuseServerMu.Lock(). Also, do you need to set c.fuseServer to nil after the unmount is complete? 

To do double-check lock correctly, you'd have to do this: 
```
c.fuseServerMu.Lock()
hasFuseServer := c.fuseServer != nil
c.fuseServerMu.Unlock()

if hasFuseServer {
  c.fuserServerMu.Lock() 
  if  c.fuseServer != nil {
     c.fuseServer.Unmount()
     c.fuseServer = nil
  }
  c.fuseServerMu.Unlock()
  // ...
}
```

Or simplify with a single-check lock pattern:
```
hasFuseServer := false
c.fuseServerMu.Lock() 
if  hasFuseServer  = c.fuseServer != nil ; hasFuseServer {
   c.fuseServer.Unmount()
   c.fuseServer = nil
}
c.fuseServerMu.Unlock()
  
if hasFuseServer {
 // ...
}
```

Reply:
Good catch. This PR is broken on Windows, and so I have one more patch to apply here. In that next patch, I just wrapped the call to `Unmount` with a mutex as you suggest.

I'll send that change as a separate commit here.